Result of `nixpkgs-review pr 345928` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>easycrypt</li>
    
```
error: builder for '/nix/store/ps2lx3kyf38g22vvfn34d22pv8v791a0-easycrypt-2024.09.drv' failed with exit code 2;
       last 18 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/6f00w85dwqv0394b4icry0brq7jk5gpk-source
       > source root is source
       > Running phase: patchPhase
       > substituteStream() in derivation easycrypt-2024.09: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'dune-project')
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > build flags: SHELL=/nix/store/iafzjk7zbkqaszqfp6n006vvxjrpn4f6-bash-5.2p32/bin/bash
       > rm -f src/ec.exe ec.native
       > dune build
       > File "src/dune", line 22, characters 7-9:
       > 22 |  (name ec)
       >             ^^
       > Error: Program codesign not found in the tree or in PATH
       > Hint: codesign should be part of the macOS installation
       > make: *** [Makefile:32: build] Error 1
       For full logs, run 'nix log /nix/store/ps2lx3kyf38g22vvfn34d22pv8v791a0-easycrypt-2024.09.drv'.
error: 1 dependencies of derivation '/nix/store/fpcxkcj1x2kq3pdxvycik2whqvb7hhyk-review-shell.drv' failed to build
```
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>easycrypt-runtest</li>
  </ul>
</details>

--- a/pkgs/applications/science/logic/easycrypt/default.nix
+++ b/pkgs/applications/science/logic/easycrypt/default.nix
@@ -1,14 +1,14 @@
-{ lib, stdenv, fetchFromGitHub, ocamlPackages, why3, python3 }:
+{ lib, stdenv, darwin, fetchFromGitHub, ocamlPackages, why3, python3 }:
 
 stdenv.mkDerivation rec {
   pname = "easycrypt";
-  version = "2024.01";
+  version = "2024.09";
 
   src = fetchFromGitHub {
     owner = pname;
     repo = pname;
     rev = "r${version}";
-    hash = "sha256-UYDoVMi5TtYxgPq5nkp/oRtcMcHl2p7KAG8ptvuOL5U=";
+    hash = "sha256-ZGYklG1eXfytRKzFvRSB6jFrOCm1gjyG8W78eMve5Ng=";
   };
 
   nativeBuildInputs = with ocamlPackages; [
@@ -17,10 +17,12 @@ stdenv.mkDerivation rec {
     menhir
     ocaml
     python3.pkgs.wrapPython
-  ];
+  ] ++ lib.optional stdenv.hostPlatform.isDarwin darwin.sigtool;
+
   buildInputs = with ocamlPackages; [
     batteries
     dune-build-info
+    dune-site
     inifiles
     why3
     yojson
@@ -32,7 +34,7 @@ stdenv.mkDerivation rec {
   strictDeps = true;
 
   postPatch = ''
-    substituteInPlace dune-project --replace '(name easycrypt)' '(name easycrypt)(version ${version})'
+    substituteInPlace dune-project --replace-fail '(name easycrypt)' '(name easycrypt)(version ${version})'
   '';
 
   pythonPath = with python3.pkgs; [ pyyaml ];

